### ISSUE TITLE
Layer Normalization

### ISSUE THREAD

**Issue by SerialLain3170** [SerialLain3170/nagato#80]
## Overview

- Instead of normalizing per channel like Batch Normalization, authors propose Layer Normalization, which normalizes per layer
  - Layer Normalization does not depend of batch size and thus is useful for RNN
  - This normalization method is invariance to weight matrix re-centering and single training case re-scaling
- Thank to the analysis of Fisher information matrix under the normalized multi-dimensional GLM with the gain parameters, authors hypothesize that the norm of the weight vector effectively controls the learning rate for the weight vector
 
![Screen Shot 2021-11-06 at 22 02 20](https://user-images.githubusercontent.com/32360147/140610525-0faac5dc-4746-4644-9152-7f5094024b4e.png)

**Comment by SerialLain3170** [comment]
## Evaluation
- With Layer Normalization, LSTM converges faster than vanilla LSTM and LSTM with Batch Normalization

![Screen Shot 2021-11-06 at 22 17 08](https://user-images.githubusercontent.com/32360147/140611071-b8d8ba90-7fbc-4615-b9ba-16347b774ac5.png)

- Layer Normalization with small batch size helps stabilize training while Batch Normalization with small batch size can be unstable

![Screen Shot 2021-11-06 at 22 18 20](https://user-images.githubusercontent.com/32360147/140611110-fc223f4c-1c5d-4ce7-9c3d-f1b30d294798.png)

**Comment by SerialLain3170** [comment]
## Supplemental Information

[Paper Link](https://arxiv.org/pdf/1607.06450.pdf)

### Authors
Jimmy Lei Ba, Jamie Ryan Kiros, Geoffrey E. Hinton

### Organization
University of Toronto
Google Inc.

### Conference
NeurIPS 2016
